src/main/java/org/minions/devfund/damianvp/WhackAMole.java

```diff
@@ -0,0 +1,184 @@
+
+    /**
+     * Method to place a specific mole by Location x, y.
+     * @param x  x location of grid.
+     * @param y  y location of grid.
+     * @return  true, if place is successful.
+     *          false, if place was not performed.
+     */
+    boolean place(int x, int y) {
+        if (validatedCoordinates(x, y) && this.moleGrid[x][y] != MOLE) {
+            this.moleGrid[x][y] = MOLE;
+            this.molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Method to randomly place n quantity of moles.
+     * @param moles quantity of moles to place randomly.
+     */
+    void randomPlace(int moles) {
+        Random random = new Random();
+        int max = this.moleGrid.length;
+        int min = 0;
+        boolean moleIsPlaced;
+        int limitGrid = this.moleGrid.length * this.moleGrid.length;
+        if (moles <= limitGrid) {
+            while (moles > 0) {
+                int x = random.nextInt(max - min);
+                int y = random.nextInt(max - min);
+                moleIsPlaced = place(x, y);
+                if (moleIsPlaced) {
+                    moles--;
+                }
+            }
+        }
+        LOGGER.log(Level.WARNING, "Is not possible place more than {0} moles ", limitGrid);
+    }
+
+    /**
+     * Method to whack a location in order to search a mole.
+     * @param x x location of grid game.
+     * @param y y location of grid game
+     */
+    void whack(int x, int y) {
+        if (validatedCoordinates(x, y)) {
+            if (this.moleGrid[x][y] == MOLE) {
+                LOGGER.log(Level.FINE, "Good, you found a mole!!");
+                this.molesLeft--;
+                this.score++;
+
+                this.trackerMoleGrid[x][y] = WHACKED;
+                this.moleGrid[x][y] = WHACKED;
+            }
+            this.attemptsLeft--;
+            verifyGameStatus();
+        }
+    }
+
+    /**
+     * Method to print the grid available for user.
+     */
+    void printGridToUser() {
+        String gridPrintFormat = Arrays.deepToString(this.trackerMoleGrid);
+        LOGGER.log(Level.INFO, gridPrintFormat);
+    }
+
+    /**
+     * Method to print the complete grid game status.
+     */
+    void printGrid() {
+        String gridPrintFormat = Arrays.deepToString(this.moleGrid);
+        LOGGER.log(Level.INFO, gridPrintFormat);
+    }
+
+    /**
+     * Method to validate locations.
+     * @param x x location of grid game.
+     * @param y y location of grid game.
+     * @return true, if locations are valid.
+     *         false, if location are not valid.
+     */
+    boolean validatedCoordinates(int x, int y) {
+        if (x > INVALID_RANGE && y > INVALID_RANGE && x < this.moleGrid.length && y < this.moleGrid.length) {
+            if (x == -1 && y == -1) {
+                this.attemptsLeft = 0;
+                verifyGameStatus();
+                return false;
+            }
+            return true;
+        } else {
```

Comment:
You can remove the else and just put the Logger and return false after the if condition.